Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp

```diff
@@ -190,7 +223,13 @@ void ProcessLauncher::terminateProcess()
     if (!m_processIdentifier)
         return;
 
+#if USE(LIBWPE)
+    if (WPEProcessProvider::singleton().isEnabled())
+        WPEProcessProvider::singleton().kill(m_processIdentifier);
+    else
+#endif
     kill(m_processIdentifier, SIGKILL);
```

Comment:
Carlos prefers a little code duplication rather than preprocessor guards that modify control flow:

```suggestion
#if USE(LIBWPE)
    if (WPEProcessProvider::singleton().isEnabled())
        WPEProcessProvider::singleton().kill(m_processIdentifier);
    else
        kill(m_processIdentifier, SIGKILL);
#else
    kill(m_processIdentifier, SIGKILL);
#endif
```